oscore-cf/src/main/java/org/eclipse/californium/oscore/OSException.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Joakim Brorsson
+ *    Tobias Andersson (RISE SICS)
+ *    
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+public class OSException extends Exception {
```

Comment:
Maybe extended by a `CoapOsException` with an additional response code?

Reply:
I have not heard about Coap0sException. But if it looks better I can make the change.

Reply:
> I have not heard about Coap0sException. 

Sure, the intention is, that you the one to create it :-).

Idea:

```
public class CoapOsException extends OSException {

	private final int responseCode;

	/**
	 * Constructor
	 * 
	 * @param message the message
     * @param coapResponseCode coap response code for error response
	 */
	public CoapOsException(String message, int coapResponseCode) {
		super(message);
		this.responseCode = coapResponseCode;
	}
	
	public int getResponseCode() {
	    return responseCode;
	}
}
```

Your solution uses the "message" to map the response code, I would prefer a specialized exception for that.

Reply:
Ha :). I will make it happen.